src/mongo/db/catalog/database_impl.cpp

```diff
@@ -600,8 +600,10 @@ Status DatabaseImpl::dropCollectionEvenIfSystem(OperationContext* opCtx,
 
     // Register this drop-pending namespace with DropPendingCollectionReaper to remove when the
     // committed optime reaches the drop optime.
-    repl::DropPendingCollectionReaper::get(opCtx)->addDropPendingNamespace(dropOpTime, dpns);
-
+    auto &&pReaper = repl::DropPendingCollectionReaper::get(opCtx);
```

Comment:
A running mongod server must always have a value DropPendingCollectionReaper. Can you see which tests are failing to provide a valid DropPendingCollectionReaper instance?